src/all/unionmangas/src/eu/kanade/tachiyomi/extension/all/unionmangas/UnionMangas.kt

```diff
@@ -0,0 +1,231 @@
+package eu.kanade.tachiyomi.extension.all.unionmangas
+
+import android.util.Log
+import eu.kanade.tachiyomi.lib.cryptoaes.CryptoAES
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.asObservableSuccess
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+import uy.kohesive.injekt.injectLazy
+import java.security.MessageDigest
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import java.util.TimeZone
+
+class UnionMangas(
+    private val langOption: LanguageOption,
+) : ParsedHttpSource() {
+    override val lang = langOption.lang
+
+    override val name: String = "Union Mangas"
+
+    override val baseUrl: String = "https://unionmangas.xyz"
+
+    override val supportsLatest = true
+
+    private val json: Json by injectLazy()
+
+    private fun apiHeaders(url: String): Headers {
+        val date = SimpleDateFormat("EE, dd MMM yyyy HH:mm:ss 'GMT'", Locale.ENGLISH)
```

Comment:
This is the default behavior of the page, sending a new date and generating a new hash with each request in the header.

Computing the dynamic MD5 for **__hash_** and **__tranId_** is necessary, otherwise it returns 401

Reply:
I man the `SimpleDateFormat`, you will still use it to generate new date

Reply:
Ok, I thought you were talking about the apiHeaders function